Frontend/ui-customer/src/Screens/ZooTicketBookingFlow/TicketBookingScreen/View.purs

```diff
@@ -293,244 +365,696 @@ feeBreakUpView state push services =
+                let startTime = if sti.startTime /= "" then convertUTCTimeToISTTimeinHHMMSS sti.startTime else ""
+                    endTime = if sti.endTime /= "" then convertUTCTimeToISTTimeinHHMMSS sti.endTime else ""
+                if currentDate == state.data.dateOfVisit then do
+                    case expiry of
+                      API.InstantExpiry val -> do
+                        if startTime /= "" then do
+                          let newStartTime = runFn3 incrOrDecrTimeFrom startTime val false
+                          if (now > newStartTime) then 
+                              if (endTime /= "") then do
+                                if (now < endTime) then find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+                                else Nothing 
+                              else find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+                          else
+                              if (endTime /= "") then do
+                                if (now < endTime) then find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+                                else Nothing 
+                              else Nothing
+                        else Nothing
+                      _ ->  if (endTime /= "") then do
+                              if (now < endTime) then find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+                              else Nothing 
+                            else Nothing
+                else find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+        Just bhId -> find (\bh -> bh.bhourId == bhId && selOperationalDay `elem` bh.operationalDays) bhs
+
+placeClosed :: Maybe TicketPlaceResp -> Boolean
+placeClosed mbPlaceInfo = do
+  case mbPlaceInfo of
+    Nothing -> false
+    Just (TicketPlaceResp pInfo) -> do
+      let currentTime = convertUTCtoISC (getCurrentUTC "") "HH:mm:ss"
+      case pInfo.closeTimings of
+        Nothing -> false
+        Just closeTime -> case pInfo.openTimings of
+            Nothing -> false
+            Just startTime -> do
+              if (convertUTCTimeToISTTimeinHHMMSS startTime) <= currentTime && currentTime <= (convertUTCTimeToISTTimeinHHMMSS closeTime) then false else true
+
+placeClosedToday :: Maybe TicketPlaceResp -> String -> Boolean
+placeClosedToday mbPlaceInfo dateOfVisit = do
+  case mbPlaceInfo of
+    Nothing -> false
+    Just (TicketPlaceResp pInfo) -> do
+      let currentTime = convertUTCtoISC (getCurrentUTC "") "HH:mm:ss"
+          currentDate = convertUTCtoISC (getCurrentUTC "") "YYYY-MM-DD"
+      if currentDate == dateOfVisit then
+        case pInfo.closeTimings of
+          Nothing -> false
+          Just closeTime -> do
+            if currentTime <= (convertUTCTimeToISTTimeinHHMMSS closeTime) then false else true
```

Comment:
Boolean login can be simplified as 
Just closeTime -> currentTime > convertUTCTimeToISTTimeinHHMMSS closeTime

Reply:
done